app/frontend/components/shared/permit-applications/requirement-form.tsx

```diff
@@ -1,51 +1,128 @@
+import { ArrowUp } from "@phosphor-icons/react"
+import { observer } from "mobx-react-lite"
+import React, { useEffect, useRef, useState } from "react"
+import { useTranslation } from "react-i18next"
+import { useNavigate } from "react-router-dom"
+import { useMountStatus } from "../../../hooks/use-mount-status"
 import { IPermitApplication } from "../../../models/permit-application"
+import { handleScrollToTop } from "../../../utils/utility-funcitons"
 import { Form } from "../chefs"
 
 interface IRequirementFormProps {
   permitApplication?: IPermitApplication
-  requirements?: any
+  onFormChange: (submission: any) => void
 }
 
-export function RequirementForm({ permitApplication, requirements }: IRequirementFormProps) {
-  const form = permitApplication ? permitApplication.requirements : requirements
-  const submission = permitApplication?.submissionData
+export const RequirementForm = observer(({ permitApplication, onFormChange }: IRequirementFormProps) => {
+  const { submissionData, setSelectedTabIndex, indexOfBlockId, formJson, blockClasses } = permitApplication
+  const isMounted = useMountStatus()
+  const { t } = useTranslation()
+  const navigate = useNavigate()
 
-  //if there is no permit application provdided, you cannto run the key items like submit OR file upload.
-  const onSubmit = (submission: any) => {
-    if (permitApplication) {
-      permitApplication.update({ submissionData: submission })
-      //error on saving should be handled by error pipeline
+  const boxRef = useRef<HTMLDivElement>(null)
+
+  const [wrapperClickCount, setWrapperClickCount] = useState(0)
+
+  const [allCollapsed, setAllCollapsed] = useState(false)
+
+  const togglePanelCollapse = () => {
+    if (!allCollapsed) {
+      document.querySelectorAll(".formio-collapse-icon.fa-minus-square-o").forEach((el: HTMLDivElement) => el.click())
     } else {
-      alert("This is a sample render, you can only submit a real permit applicaiton.")
+      document.querySelectorAll(".formio-collapse-icon.fa-plus-square-o").forEach((el: HTMLDivElement) => el.click())
     }
+    setAllCollapsed((cur) => !cur)
   }
 
-  if (!permitApplication && !requirements) {
-    return (
-      <Alert status="error">
-        <AlertIcon />
-        <AlertTitle>You are not rendering a Permit Application or Requirement Preview</AlertTitle>
-        <AlertDescription>Please have </AlertDescription>
-      </Alert>
+  useEffect(() => {
+    // The box observers need to be re-registered whenever a panel is collapsed
+    // This triggers a re-registration whenever the body of the box is clicked
+    // Click listender must be registered this way because formIO prevents bubbling
+
+    const box = boxRef.current
+    const handleClick = () => {
+      setWrapperClickCount((n) => n + 1)
+    }
+    box?.addEventListener("click", handleClick)
+    return () => {
+      box?.removeEventListener("click", handleClick)
+    }
+  }, [])
```

Comment:
Why does this need to be done imperatively in an effect. I think this can be added to as an onClick to the box directly?

Reply:
formIO prevents bubbling, this gets around it somehow